cluster-autoscaler/core/scaledown/legacy/wrapper.go

```diff
@@ -81,6 +81,12 @@ func (p *ScaleDownWrapper) NodeUtilizationMap() map[string]utilization.Info {
 
 // StartDeletion triggers an actual scale down logic.
 func (p *ScaleDownWrapper) StartDeletion(empty, needDrain []*apiv1.Node, currentTime time.Time) (*status.ScaleDownStatus, errors.AutoscalerError) {
+	_, drained := p.sd.nodeDeletionTracker.DeletionsInProgress()
```

Comment:
Have you thought about putting this part inside `TryToScaleDown`? Maybe then after extracting the actuation logic from `TryToScaleDown`, `TryToScaleDown` could become `NodesToDelete`, and we could get rid of this wrapper?

Reply:
Moving to `TryToScaleDown` sounds good (done), but I'm not sure we can easily make it into `NodesToDelete` and remove the wrapper, since `TryToScaleDown` has a lot of logic for setting the scale down status which won't make sense in `NodesToDelete`. We'll still need the wrapper (although not necessarily this whole file) around the common actuation part.